refine_dictionaries/get_closest_words.py

```diff
@@ -0,0 +1,373 @@
+  core_df = pd.DataFrame([core, cos_avg_sims_core_all,  top3_tuple_lst_core_all]).T
+  top_500_df = pd.DataFrame([top_500_all, cos_avg_sims_top_500_all,  top3_tuple_lst_top_500_all]).T
+  df_concat = pd.concat([core_df, top_500_df])
+  df_concat.columns = ['term', 'cosine_sim_with_core', ' closest_terms_from_core']
+  return df_concat
+
+def get_results_decade(model, cult_core, dem_core, relt_core, cult_dict, dem_dict, relt_dict, get_500=False):
+    
+    """
+    For each perspective, get dataframe containing all the words in the list top_500_all, their average cosine similarity with the core dict and the closest 3 terms from the core dict
+    
+    input:
+        - model = w2v model
+        - cult_core(list of str) = list of core words of cultural perspective
+        - dem_core (list of str)= list of core words of demographic perspective
+        - relt_core (list of str)= list of core words of relational perspective
+        - cult_dict (list of str)= list of refined terms for cultural perspective, 
+            could be empty list when getting similarities of expanded dicts 
+            (500 terms closest core terms)
+        - dem_dict (list of str)= list of refined terms for demographic perspective, 
+            could be empty list when getting similarities of expanded dicts 
+            (500 terms closest core terms)   
+        - relt_dict (list of str)= list of refined terms for relational perspective, 
+            could be empty list when getting similarities of expanded dicts 
+            (500 terms closest core terms)   
+        - get_500 (bool)= boolean indicating whether to use generated 500 expanded similar terms instead of using refined terms
+   
+   outputs:
+       - cult_concat (pandas df) = cultural dataframe with the columns 'term', 'cosine_sim_with_core', ' closest_terms_from_core'
+       - dem_concat (pandas df)= demographic dataframe with the columns 'term', 'cosine_sim_with_core', ' closest_terms_from_core'
+       - relt_concat (pandas df) = relational dataframe with the columns 'term', 'cosine_sim_with_core', ' closest_terms_from_core'
+    
+    """
+    # remove terms from each dictionary not in w2v model vocab
+    
+    dem_core=remove_terms(dem_core, model)
+    cult_core=remove_terms(cult_core, model)
+    relt_core=remove_terms(relt_core, model)
+    
+    if get_500:
+        cult_dict = [i[0] for i in model.wv.most_similar(cult_core, topn = 500)]
+        dem_dict = [i[0] for i in model.wv.most_similar(dem_core, topn = 500)]
+        relt_dict = [i[0] for i in model.wv.most_similar(relt_core, topn = 500)]
+    
+    cult_concat=get_output_df(cult_dict, cult_core,model)
+    dem_concat=get_output_df(dem_dict, dem_core,model)
+    relt_concat=get_output_df(relt_dict, relt_core,model)
+    return cult_concat, dem_concat, relt_concat
+
+
+
+####################################
+###### get similarities of expanded dicts (500)
+####################################
+
+def save_expanded_dicts_df():
```

Comment:
Defining functions that execute the main functions and saves the output makes sense to me given you pass in the option of refining or expanding the dictionaries. However, the function names should make more clear what they are doing (not just saving), something more like below (drop the `df` part for simplicity). 

```suggestion
def get_results_save_expanded():
```